runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/resolver/ExpressionResolver.java

```diff
@@ -0,0 +1,49 @@
+package io.aklivity.zilla.runtime.engine.resolver;
+
+import static java.util.Collections.unmodifiableMap;
+import static java.util.Objects.requireNonNull;
+import static java.util.ServiceLoader.load;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.ServiceLoader;
+
+public class ExpressionResolver
+{
+    private final Map<String, ExpressionResolverSpi> resolverSpis;
+
+    public static ExpressionResolver instantiate()
+    {
+        return instantiate(load(ExpressionResolverSpi.class));
+    }
+
+    public String resolve(
+            String name,
+            String config)
```

Comment:
Just a single argument, `config` needed here.

Applying the pattern to the `config` allows us to find each occurrence of `${{ context.var }}` and then lookup the resolver SPI for `context` to resolve `var`.